src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -26,21 +26,24 @@ final class DateTimeNormalizer implements NormalizerInterface, DenormalizerInter
     public const TIMEZONE_KEY = 'datetime_timezone';
     public const CAST_KEY = 'datetime_cast';
 
+    public const DEFAULT_SUPPORTED_TYPES = [
+        \DateTimeInterface::class => true,
+        \DateTimeImmutable::class => true,
+        \DateTime::class => true,
+    ];
+
     private array $defaultContext = [
         self::FORMAT_KEY => \DateTimeInterface::RFC3339,
         self::TIMEZONE_KEY => null,
         self::CAST_KEY => null,
     ];
 
-    private const SUPPORTED_TYPES = [
-        \DateTimeInterface::class => true,
-        \DateTimeImmutable::class => true,
-        \DateTime::class => true,
-    ];
+    private array $supportedTypes;
 
-    public function __construct(array $defaultContext = [])
+    public function __construct(array $defaultContext = [], ?array $supportedTypes = null)
```

Comment:
```suggestion
    /**
     * @param null|string[] $supportedTypes
     */
    public function __construct(array $defaultContext = [], ?array $supportedTypes = null)
```

This will improve static analysis and readability.

Reply:
@dunglas committed. Sorry for not doing "Commit suggestion", I realized late this this is a new GitHub feature.

Reply:
No worries! Making the change by hand is fine and often more reliable (coding in the GitHub UI is not very comfortable).